src/main/java/com/example/videotecha/repository/ReservationRepository.java

```diff
@@ -0,0 +1,20 @@
+package com.example.videotecha.repository;
+
+import com.example.videotecha.model.Reservation;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+import java.util.List;
+
+public interface ReservationRepository extends JpaRepository<Reservation, Long> {
+
+    @Query("SELECT count(r.id) " +
+            "FROM Reservation r " +
+            "WHERE r.projection.id = :projection_id and r.user.id = :user_id ")
```

Comment:
You can use triple quotes for better readability here, like this

"""
my
multline
string
"""

Reply:
Also not a bad idea to add newlines between where clauses, easier to read

Reply:
Also, this can be done via query-by-method-name. I think `countByUserAndReservation` or `countByUserIdAndReservationId`